data/list/basic.lean

```diff
@@ -957,6 +957,32 @@ theorem drop_eq_nth_le_cons : ∀ {n} {l : list α} h,
 | 0     (a::l) h := rfl
 | (n+1) (a::l) h := @drop_eq_nth_le_cons n _ _
 
+/- simp? -/
+theorem drop_drop : ∀ (n m) (l : list α), drop n (drop m l) = drop (n + m) l
+| n     0     l      := by simp
+| 0     (m+1) l      := by simp
+| (n+1) (m+1) nil    := by simp [drop_nil]
+| (n+1) (m+1) (a::l) :=
+  have h: n + (m + 2) = (n + (m + 1)) + 1, by simp,
+  show drop (n + 1) (drop (m + 1) (a :: l)) = drop (n + 1 + (m + 1)) (a :: l),
+  begin 
+    simp [drop_drop, drop], 
+    rw [h], 
+    refl 
+  end
+
+/- simp? -/
+theorem drop_take : ∀ (m : ℕ) (n : ℕ) (l : list α),
```

Comment:
I don't think this should be a `simp`-rule, the `m+n` is not very predicable.
We would need to decide what (term) order we want to have on `drop` and `take` for our rewrite rules.